sql/analyzer/resolve_columns.go

```diff
@@ -19,6 +19,11 @@ type deferredColumn struct {
 	*expression.UnresolvedColumn
 }
 
+// IsNullable implements the Expression interface.
+func (deferredColumn) IsNullable() bool {
```

Comment:
If deferred column should implement Expression interface, can you add a dummy var forcing it? (`var _ Expression = deferredColumn{}`)

Reply:
But it already implements it through `*expression.UnresolvedColumn`. I've just overwrote `IsNullable`.

Reply:
I don't know if this is the right solution, though. Technically, since the column has not been resolved we cannot tell for sure if is nullable or not. Maybe we should leave a comment or something behind the rationale of this?

Reply:
So the question is when we mark column as _deferred_? In theory when we cannot resolve it, so I assumed that it's not primitive value - that's why I return true.

On the other hand - doesn't really matter for deferred column is it `nullable` or not?

Reply:
I don't think it matters much because it's not used directly, so let's keep it this way.